### ISSUE TITLE
Architecture issue

### ISSUE THREAD

**Issue by minhnimble** [iamdarqueza/nimsurvey#4]
One concerning thing which is obvious to see is that you are using MVC as the main architecture for the application. Why do you choose to implement the application using MVC and are you aware of the advantages and disadvantages of the MVC architecture?

It was quite noticeable that there are quite a few improvements can be made on your Service layer. As of now, all the network request handlers are implemented manually as functions in the `BaseService` class and no Network layer at all. Would you consider to implement a Network layer and utilize the interceptor/authenticator from the Alamofire library to help the application adapt and retry the requests with more scalability, leading to effortless new extensions in the future like force-upgrade, force-logout, header adaption and ssl pinning down the road in each request?

For the data layer, I noticed that you are using Codable and parse JSON API formatted responses manually instead of using a 3rd-party library for handling this task. Why do you think parsing manually like this would be a better choice? And do you think using a JSON API parser library would add more benefits?
https://github.com/iamdarqueza/nimsurvey/blob/621c7137d6d17dcd07f2c24a0ddecb230cc996ca/Nimble/Model/SurveyList.swift#L11-L49

**Comment by iamdarqueza** [comment]
Yes, I'm more familiar with MVC that's why I used the framework but I'm slowly migrating to MVVM but chose this architecture for this one since it's a a small project. But can also code this one on MVVM but might not be as fast as on MVC since im still on the learning side
